File: game_server/ClientProtocol/GS2GC/p004_PlayerOp/GS2GC_004_012_RetTitleRecordList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p004_PlayerOp
{

public class GS2GC_004_012_RetTitleRecordList : ALBasicProtocolPack._IALProtocolStructure {
private List<long> titleIdList;


public GS2GC_004_012_RetTitleRecordList() {
	titleIdList = new List<long>();
}

public GS2GC_004_012_RetTitleRecordList(
	List<long> _titleIdList
) {	titleIdList = _titleIdList;
}

public byte getMainOrder() { return (byte)4; }

public byte getSubOrder() { return (byte)12; }

public List<long> getTitleIdList() { return titleIdList; }
public void addTitleIdList(long _titleIdList) { titleIdList.Add(_titleIdList); }


public int GetBufSize() {
	int _size = 0;
	_size += 2 + (titleIdList.Count * 8);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (titleIdList.Count * 8);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _titleIdListCount = _buf.getShort();
	for(int _i = 0; _i < _titleIdListCount; _i++) { 
		long _titleIdList = (long)0;
		_titleIdList = _buf.getLong();
		titleIdList.Add(_titleIdList);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)titleIdList.Count);
	for(int _i = 0; _i < titleIdList.Count; _i++) { 
		_buf.putLong(titleIdList[_i]);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)12);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)12);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("titleIdList").Append(":").Append(titleIdList.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GS2GC/p004_PlayerOp/GS2GC_004_012_RetTitleRecordList.java
package GS2GC.p004_PlayerOp;

import java.nio.ByteBuffer;
public class GS2GC_004_012_RetTitleRecordList implements ALBasicProtocolPack._IALProtocolStructure {
private java.util.ArrayList<Long> titleIdList;


public GS2GC_004_012_RetTitleRecordList() {
	titleIdList = new java.util.ArrayList<Long>();
}

public GS2GC_004_012_RetTitleRecordList(
	 java.util.ArrayList<Long> _titleIdList
) {	titleIdList = _titleIdList;
}

public final byte getMainOrder() { return (byte)4; }

public final byte getSubOrder() { return (byte)12; }

public java.util.ArrayList<Long> getTitleIdList() { return titleIdList; }
public void addTitleIdList(long _titleIdList) { titleIdList.add(_titleIdList); }


public final int GetBufSize() {
	int _size = 0;
	_size += 2 + (titleIdList.size() * 8);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (titleIdList.size() * 8);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _titleIdListCount = _buf.getShort();
	for(int _i = 0; _i < _titleIdListCount; _i++) { 
		long _titleIdList = (long)0;
		if(_buf.remaining() > 0) _titleIdList = _buf.getLong();
		titleIdList.add(_titleIdList);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)titleIdList.size());
	for(int _i = 0; _i < titleIdList.size(); _i++) { 
		_buf.putLong(titleIdList.get(_i));
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)12);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)12);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

